src/machine/machinectl.c

```diff
@@ -1420,6 +1496,10 @@ static int shell_machine(int argc, char *argv[], void *userdata) {
                 return bus_log_create_error(r);
 
         r = sd_bus_call(bus, m, 0, &error, &reply);
+        if (r == -EOPNOTSUPP) {
```

Comment:
while this would work I think it would be nicer to check the dbus error here, ie. use sd_bus_error_has_name()...

the errno we return is a lot less precise, and given that there is no need to be imprecise here we should use the bus error.